### ISSUE TITLE
[alpm] adopt v5.0.0 changes

### ISSUE THREAD

**Issue by philmmanjaro** [manjaro/pamac#90]
Archlinux just released [pacman 5.0.0](https://lists.manjaro.org/pipermail/manjaro-dev/Week-of-Mon-20160125/000898.html). Since we use **alpm** for **pamac**, we have already started to adopt to these new changes:

> I'm currently working on some other improvements and the support of pacman 5.
>
> Regards,
> Guillaume.

**Comment by lots0logs** [comment]
@philmmanjaro What's the status on this? If no one is working on it already please let me know and I will review it and submit a PR with necessary changes.

**Comment by philmmanjaro** [comment]
Hi @lots0logs, I've just the status @guinux gave me. I'm actually don't know the current status yet, so I've to still wait for a feedback from Guillaume. I'm also happy to adopt any given merge request for this matter. I also know that Arch already ships the new 5.0 version also in stable.

**Comment by philmmanjaro** [comment]
Followed changes I see so far in **alpm**:

`ALPM_EVENT_PACORIG_CREATED` removed, `ALPM_EVENT_HOOK_START`, `ALPM_EVENT_HOOK_DONE`, `ALPM_EVENT_HOOK_RUN_START`, `ALPM_EVENT_HOOK_RUN_DONE` added. Compiling **pamac** might result in `Error: »ALPM_SIG_PACKAGE_SET« not declared`

**Comment by philmmanjaro** [comment]
Here we go for the full list of changes in ABI:
```
API CHANGES BETWEEN 4.2 AND 5.0
===============================

[REMOVED]
- alpm_siglevel_t - removed members ALPM_SIG_PACKAGE_SET, ALPM_SIG_PACKAGE_TRUST_SET
- removed .pacorig generation
  - ALPM_EVENT_PACORIG_CREATED
  - alpm_event_pacorig_created_t
  - alpm_event_t.pacorig_created

[ADDED]
- hook support
  - alpm_option_get_hookdirs()
  - alpm_option_set_hookdirs()
  - alpm_option_add_hookdir()
  - alpm_option_remove_hookdir()
  - alpm_event_hook_t, alpm_event_hook_run_t
  - alpm_hook_when_t
  - ALPM_EVENT_HOOK_START, ALPM_EVENT_HOOK_DONE
  - ALPM_EVENT_HOOK_RUN_START, ALPM_EVENT_HOOK_RUN_DONE
  - ALPM_ERR_TRANS_HOOK_FAILED
- different database extension support
  - alpm_option_get_dbext()
  - alpm_option_set_dbext()
- pkgbase accessor
  - alpm_pkg_get_base()
- transaction events
  - ALPM_EVENT_TRANSACTION_START, ALPM_EVENT_TRANSACTION_DONE
- database unlocking
  - alpm_unlock()```

**Comment by philmmanjaro** [comment]
However, we should use **alpm_unlock** for some used cases ...

**Comment by Zeph33** [comment]
I create simple patch for build with pacman5, I do not guarantee that everything works
I have check all type enum in libalpm.vapi and comment all removed
https://aur.archlinux.org/cgit/aur.git/tree/pacman5.patch?h=pamac-aur

**Comment by philmmanjaro** [comment]
Will check, however this is the news/status I've from Guillaume now:

> Pamac is almost ready but we need to keep pacman 4 until I polish it.
> Guillaume.

**Comment by guinux** [comment]
Fixed in v3.0.0
